TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.BasicBlockForNode.cs

```diff
@@ -0,0 +1,42 @@
+﻿using TypeCobol.Analysis.Graph;
+using TypeCobol.Compiler.CodeElements;
+using TypeCobol.Compiler.Nodes;
+
+namespace TypeCobol.Analysis.Cfg
+{
+    public partial class ControlFlowGraphBuilder<D>
+    {
+        /// <summary>
+        /// BasicBlock which instruction are Nodes.
+        /// </summary>
+        /// <typeparam name="D"></typeparam>
+        public class BasicBlockForNode : BasicBlock<Node, D>
+        {
+            /// <summary>
+            /// Any tag associated to this Node.
+            /// </summary>
+            public object Tag
```

Comment:
This is used to memorize the section or paragraph name in which this block appears, so this can be turned into `string ProcedureName`.

Reply:
This is a general tag having object type, I don't want to give it as meaning ProcedureName out of any context.

Reply:
Well actually, the `Tag` property is defined for the derived class `BasicBlockForNode` so it already acknowledges a Cobol (or TypeCobol) context because it uses the `Node` class to describe instructions. In Cobol context I think it makes sense to describe a block with the section or paragraph in which it appears.